src/lib.rs

```diff
@@ -207,16 +222,83 @@ impl Encode for Module {
     }
 }
 
+#[cfg_attr(test, derive(Clone, Debug, Eq, PartialEq))]
+pub enum CENNZnut {
+    V0(CENNZnutV0),
+}
+
+impl Encode for CENNZnut {
+    fn encode_to<T: Output>(&self, buf: &mut T) {
+        use CENNZnut::*;
+
+        match &self {
+            V0(inner) => inner.encode_to(buf),
+        }
+    }
+}
+
+impl Decode for CENNZnut {
+    fn decode<I: Input>(input: &mut I) -> Result<Self, codec::Error> {
+        use CENNZnut::*;
```

Comment:
are these local imports needed?

Reply:
Yes, without it I get a compile error for V0 a few lines down.

Reply:
> This is great!
> Trait-ifying `get_module(s)` is a bit trickier than doing this since the Module version is dependent on the CENNZnut version.
> Instead, it would be good to have an `impl Into<CENNZnutV0> for CENNZnut`
> then the test
> https://github.com/cennznet/cennznut-rs/pull/35/files#diff-c5fc4beef0f03493c2e9a9c876073f0fR80 would become `c.into().modules.len()`

The conversion to V0 may be optionally none then. Without get_modules, a bigger part of the code would depend on the concrete implementation of different versions and thus would be prone to change. I guess we can define a common trait for what makes a module a module so it remains valid for future versions of Cennznut too.

Reply:
> are these local imports needed?

Yes, I get compile errors on V0 a few lines further down, otherwise.

Reply:
Can import once at the top of the file: `use crate::CENNZnut::V0;`

Reply:
> The conversion to V0 may be optionally none then

Yes, it would be a fallible conversion, so rather `impl TryInto`. The requirements for #30 are that versioned decoding works and the `Validate` trait is exposed which is already accomplished nicely in this PR. The module stuff is extra and should be delayed until there is a need for it

Reply:
Alright then, no worries. I'll update this PR soon.